### PR TITLE
Fix fileupload defaults not applied

### LINKED ISSUE #7085: File upload fails with "Cannot read property 'enableForAnonymousUser' of undefined"
### New Issue Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
Trying to upload a file (as a logged in user), the request endlessly  stales and the server errors with:
```
parse_1      | (node:1) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'enableForAnonymousUser' of undefined
parse_1      |     at createHandler (/parse-server/lib/Routers/FilesRouter.js:113:41)
parse_1      |     at Layer.handle [as handle_request] (/parse-server/node_modules/express/lib/router/layer.js:95:5)
parse_1      |     at next (/parse-server/node_modules/express/lib/router/route.js:137:13)
parse_1      |     at /parse-server/lib/middlewares.js:252:7
parse_1      |     at runMicrotasks (<anonymous>)
parse_1      |     at processTicksAndRejections (internal/process/task_queues.js:93:5)
parse_1      | (node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
```

### Steps to reproduce
Get the docker-image (no custom configuration) try to upload a file as a regular user. The promise never finishes.

### Actual Outcome
The server error's and the network never responds.

### Expected Outcome
The server should accept the upload, store the file and respond

### Failing Test Case / Pull Request
Looks like it was introduced in  #7071.

###  Environment
Using current latest docker image: https://hub.docker.com/layers/parseplatform/parse-server/latest/images/sha256-1b8dfd378cc731e9b81f6398e0eadbaef368f60ad9d731d09b299df0d375ffda?context=explore

Database
- System (MongoDB or Postgres): `Postgres`
- Database version: `FILL_THIS_OUT`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `Javascript`
- SDK version: `2.19.0`

### PR DESCRIPTION
**Pull request by mtrezza**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Default values of `fileUpload` are not applied if `fileUpload` option is not set explicitly in the Parse Server config option.

Related issue: closes #7085

### Approach
Added default definition for `fileUpload` group parameter.

### TODOs before merging

- [x] Add test cases

### CODE REVIEW

**Comment by Moumouls** [comment]
@mtrezza seems to have many change, did you run prettier and lint fix ?

**Comment by mtrezza** [comment]
@parse-community/server This PR should be reviewed ASAP, it fixed a bug in the current main branch that prevents file upload if default settings are not set.

File: src/Options/Definitions.js [hunk 1]
```diff
@@ -3,604 +3,626 @@
...
+  filesAdapter: {
+    env: 'PARSE_SERVER_FILES_ADAPTER',
+    help: 'Adapter module for the files sub-system',
+    action: parsers.moduleOrObjectParser,
+  },
+  fileUpload: {
+    env: 'PARSE_SERVER_FILE_UPLOAD_OPTIONS',
+    help: 'Options for file uploads',
+    action: parsers.objectParser,
+    default: {},
```
Comment (Moumouls) [inline src/Options/Definitions.js:182]: Any way to remove the default empty object ? An empty object can be translated to undefined.

**Review by Moumouls — COMMENTED** [review]
A lot of changes on spec/ParseFile, does prettier is ok on it ?

File: src/Options/Definitions.js (hunk 1 above)
Comment (mtrezza) [reply src/Options/Definitions.js:182]: Why do you want to remove it?

**Comment by mtrezza** [comment]
@dplewis @davimacedo can you please approve this, so we can merge this? The current master has a bug that does not allow file upload. I am afraid this may be forgotten before a new Parse Server release, so I'd rather see this merged ASAP.

**Review by dplewis — APPROVED** [review]

**Comment by dplewis** [comment]
Looks like the Parse.File.spec.js has a lot of changes. @Moumouls Also mentioned this.

**Comment by mtrezza** [comment]
Just some describe grouping, hence large diff, see https://github.com/parse-community/parse-server/commit/977e16ab3901b961342947d589458d419239dfed

File: src/Options/Definitions.js (hunk 1 above)
Comment (mtrezza) [reply src/Options/Definitions.js:182]: @Moumouls Do you want to elaborate your concern? Otherwise I’ll merge.

**Comment by dplewis** [comment]
LGTM!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**